core/arch/arm/tee/tee_rpmb.c

```diff
@@ -804,18 +841,42 @@ static TEE_Result tee_rpmb_init_read_wr_cnt(uint16_t dev_id,
 	return res;
 }
 
-/*
- * This function must never return TEE_SUCCESS if rpmb_ctx == NULL.
- */
-static TEE_Result tee_rpmb_init(uint16_t dev_id, bool writekey, bool commercial)
+static TEE_Result tee_rpmb_verify_key_sync_counter(uint16_t dev_id)
 {
+	uint16_t op_result = 0;
 	TEE_Result res = TEE_ERROR_GENERIC;
+
+	res = tee_rpmb_init_read_wr_cnt(dev_id, &rpmb_ctx->wr_cnt,
+					&op_result);
+
+	if (res == TEE_SUCCESS) {
+		rpmb_ctx->key_verified = true;
+		rpmb_ctx->wr_cnt_synced = true;
+	}
+
+	IMSG("Verify key returning 0x%x\n", res);
+	return res;
+}
+
+/* True when all the required crypto functions are available */
+static bool have_crypto_ops(void)
+{
+	return (crypto_ops.mac.init && crypto_ops.mac.update &&
+		crypto_ops.mac.final && crypto_ops.prng.read);
+}
+
+/* This function must never return TEE_SUCCESS if rpmb_ctx == NULL */
+static TEE_Result tee_rpmb_init(uint16_t dev_id)
+{
+	TEE_Result res = TEE_SUCCESS;
 	struct rpmb_dev_info dev_info;
-	uint16_t op_result;
 
-	if (rpmb_ctx == NULL) {
+	if (!have_crypto_ops())
+		return TEE_ERROR_NOT_SUPPORTED;
+
+	if (!rpmb_ctx) {
```

Comment:
`rpmb_ctx` is global. It should be protected by a mutex in case of concurrent TA. It could be done later on in another PR.